### PR TITLE
[LOG2LINES] Improve malloc() and free() uses

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

`Log2Lines` uses a mix of `malloc()` and no `malloc()`. I don't know why.
Simpler and safer code.

## Proposed changes

- Replace overkill malloc() uses
- Optimize malloc() uses
- Free all memory on exit

### CODE REVIEW

**Comment by ThFabba** [comment]
Seems like a fix without a problem.

**Comment by SergeGautherie** [comment]
Not `WIP` anymore.

File: sdk/tools/log2lines/cache.c [hunk 1]
```diff
@@ -18,8 +18,10 @@
 
 #include "log2lines.h"
 
-static char *cache_name;
-static char *tmp_name;
+static char CacheName[PATH_MAX];
+static char *cache_name = CacheName;
```
Comment (HBelusca) [inline sdk/tools/log2lines/cache.c]: Why not using directly `static char cache_name[PATH_MAX];` ? (and same remark for `tmp_name` below).

Comment (HBelusca) [reply sdk/tools/log2lines/cache.c]: oh OK the code changes the pointer.

**Comment by learn-more** [comment]
not every 'change for the sake of changing something' is an improvement.

**Comment by paulsapps** [comment]
Are you guys saying that removing unnecessary mallocs of fixed size preventing this code failing from allocation failure is not an improvement? Can't tell if sarcasm :P

File: sdk/tools/log2lines/options.c [hunk 2]
```diff
@@ -166,8 +166,8 @@ int optionParse(int argc, const char **argv)
             break;
         case 'R':
             optCount++;
-            free(opt_Revision);
-            opt_Revision = malloc(LINESIZE);
+            if (!opt_Revision)
+                opt_Revision = malloc(LINESIZE);
```
Comment (HBelusca) [inline sdk/tools/log2lines/options.c]: Note for reviewers: Now, the (already allocated) opt_Revision gets just re-used.

File: sdk/tools/log2lines/options.c [hunk 3]
```diff
@@ -96,8 +96,8 @@ int optionInit(int argc, const char **argv)
                 strcpy(opt_logFile, argv[i+1]);
                 break;
             case 'P':
-                free(opt_Pipe);
-                opt_Pipe = malloc(LINESIZE);
+                if (!opt_Pipe)
+                    opt_Pipe = malloc(LINESIZE);
```
Comment (HBelusca) [inline sdk/tools/log2lines/options.c]: Note for reviewers: Now, the (already allocated) opt_Pipe gets just re-used.

File: sdk/tools/log2lines/log2lines.c [hunk 4]
```diff
@@ -668,6 +672,18 @@ main(int argc, const char **argv)
...
         PCLOSE(dbgIn);
 
+cleanup:
+    // See optionParse().
+    if (opt_Revision)
+        free(opt_Revision);
+
+    // See optionInit().
+    if (opt_Pipe)
+        free(opt_Pipe);
```
Comment (HBelusca) [inline sdk/tools/log2lines/log2lines.c]: Reset pointer to NULL?

File: sdk/tools/log2lines/log2lines.c [hunk 5]
```diff
@@ -668,6 +672,18 @@ main(int argc, const char **argv)
     if (opt_Pipe)
         PCLOSE(dbgIn);
 
+cleanup:
+    // See optionParse().
+    if (opt_Revision)
+        free(opt_Revision);
```
Comment (HBelusca) [inline sdk/tools/log2lines/log2lines.c]: Reset pointer to NULL?

File: sdk/tools/log2lines/options.c (hunk 2 above)
Comment (Extravert-ir) [reply sdk/tools/log2lines/options.c]: I don't see the difference here - allocation size was always the same, and it is malloc and not calloc

**Pull request merged**